blocks/show-hide-group/index.js

```diff
@@ -0,0 +1,80 @@
+import { registerBlockType } from '@wordpress/blocks';
+import { InnerBlocks, useBlockProps } from '@wordpress/block-editor';
+import { dispatch, useSelect } from '@wordpress/data';
+
+// Internal dependencies.
+import metadata from './block.json';
+import './style.css';
+
+// Register the block.
+registerBlockType(metadata, {
+	edit: (props) => {
+		const blockProps = useBlockProps(); // eslint-disable-line react-hooks/rules-of-hooks
+		const { setAttributes } = props;
+		// eslint-disable-next-line react-hooks/rules-of-hooks
+		const innerBlocks = useSelect((select) => {
+			const currentBlocks = select('core/block-editor').getBlocks(
+				props.clientId
+			);
+			return currentBlocks;
+		});
+
+		// Update inner blocks' htmlId attributes when the inner blocks change.
+		let newId = '';
+		let currentIds = '';
+		for (let i = 0; i < innerBlocks.length; i++) {
+			newId = 'show-hide-section-' + i;
+			currentIds += ' ' + newId;
+			dispatch('core/block-editor').updateBlockAttributes(
+				innerBlocks[i].clientId,
+				{ htmlId: newId }
+			);
+		}
+
+		// Save the number of inner blocks and the list of IDs the Toggle All button controls.
+		const currentCount = innerBlocks.length;
+		setAttributes({ blockCount: currentCount, allIds: currentIds });
+
+		return (
+			<div {...blockProps}>
+				<InnerBlocks
```

Comment:
I think we should display the `Open all` button in the editor as well, so it's not a surprise when users visit the front-end. (If we wanted to be extra nice, we could even give users the ability to change the `open` and `close` text.)

Reply:
Good thought! Commit pushed. :)